v2/pkg/external/customtemplates/azure_blob.go

```diff
@@ -0,0 +1,121 @@
+package customtemplates
+
+import (
+	"bytes"
+	"context"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+	"github.com/projectdiscovery/gologger"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type customTemplateAzureBlob struct {
+	azureBlobClient *azblob.Client
+	containerName   string
+}
+
+func getAzureBlobClient(tenantID string, clientID string, clientSecret string, serviceURL string) (*azblob.Client, error) {
+	// Create an Azure credential using the provided credentials
+	credentials, err := azidentity.NewClientSecretCredential(tenantID, clientID, clientSecret, nil)
+	if err != nil {
+		gologger.Error().Msgf("Invalid Azure credentials: %v", err)
+		return nil, err
+	}
+
+	// Create a client to manage Azure Blob Storage
+	client, err := azblob.NewClient(serviceURL, credentials, nil)
+	if err != nil {
+		gologger.Error().Msgf("Error creating Azure Blob client: %v", err)
+		return nil, err
+	}
+
+	return client, nil
+}
+
+func (bk *customTemplateAzureBlob) Download(location string, ctx context.Context) {
+	// Set an incrementer for the number of templates downloaded
+	var templatesDownloaded = 0
+
+	// Define the local path to which the templates will be downloaded
+	downloadPath := filepath.Join(location, CustomAzureTemplateDirectory, bk.containerName)
+
+	// Get the list of all templates from the container
+	pager := bk.azureBlobClient.NewListBlobsFlatPager(bk.containerName, &azblob.ListBlobsFlatOptions{
+		// Don't include previous versions of the templates if versioning is enabled on the container
+		Include: azblob.ListBlobsInclude{Snapshots: false, Versions: false},
+	})
+
+	// Loop through the list of blobs in the container and determine if they should be added to the list of templates
+	// to be returned, and subsequently downloaded
+	for pager.More() {
+		resp, err := pager.NextPage(context.TODO())
+		if err != nil {
+			gologger.Error().Msgf("Error listing templates in Azure Blob container: %v", err)
+			return
+		}
+
+		for _, blob := range resp.Segment.BlobItems {
+			// If the blob is a .yaml, .yml, or .json file, download the file to the local filesystem
+			if strings.HasSuffix(*blob.Name, ".yaml") || strings.HasSuffix(*blob.Name, ".yml") || strings.HasSuffix(*blob.Name, ".json") {
```

Comment:
even `.yml` is a valid extension, nuclei only load / read `.yaml` for template loading, so we can exclude `.yml` as well.

Reply:
Fixed in `cd1b9d3e`